### PR TITLE
Cnolewajka/resource links and link objects

### PR DESCRIPTION
**Pull request by KatherineJDev**

### CODE REVIEW

File: src/NJsonApiCore/NJsonApiCore.project.lock.json [hunk 1]
```diff
@@ -242,7 +242,7 @@
     ".NETFramework,Version=v4.5.2": []
   },
   "packageFolders": {
-    "C:\\Users\\cezaryn\\.nuget\\packages\\": {}
```
Comment (cnolewajka) [inline src/NJsonApiCore/NJsonApiCore.project.lock.json]: Remove .lock.json file from the change

File: test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+using System;
+using Xunit;
+using NJsonApi.Infrastructure;
+
+namespace NJsonApi.Test.Serialization.JsonApiTransformerTest
+{
+    public class TestLinkData
+    {
+        [Fact]
+        public void Creates_CompoundDocument_for_single_class_with_no_links_and_properly_map_no_links()
```
Comment (cnolewajka) [inline test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs:16]: Are we actually asserting the links in the resulting CompoundDocument or just checking it doesn't break?

File: test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            var transformer = new JsonApiTransformerBuilder()
+                .With(CreateConfiguration())
+                .Build();
+
+            // Act
+            CompoundDocument result = transformer.Transform(objectToTransform, context);
+
+            // Assert
+            var transformedObject = result.Data as SingleResource;
+            Assert.Equal("url1", ((ISimpleLink)transformedObject.Links["link1"]).Href);
```
Comment (cnolewajka) [inline test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs:51]: Could we test the transformation of LinkObject too? In the same test or in a separate test?

**Review by cnolewajka — CHANGES_REQUESTED** [review]

File: test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs (hunk 2 above)
Comment (KatherineJDev) [reply test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs:16]: Added check for self link

File: test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs (hunk 3 above)
Comment (KatherineJDev) [reply test/NJsonApiCore.Test/Serialization/JsonApiTransformerTest/TestLinkData.cs:51]: Added separate test for transformation of LinkObject

**Review by cnolewajka — APPROVED** [review]

**Pull request merged**